### PR TITLE
[blocker 2.4] NEXUS-5639: Deep rename of WL into Routing

### PR DESCRIPTION
**Pull request by cstamas**
Continues the "shallow" pull
https://github.com/sonatype/nexus/pull/794

This pull does a full rename of proxy404 related classes and APIs.

Changes:
- package from org.sonatype.nexus.proxy.maven.wl to org.sonatype.nexus.proxy.maven.routing
- classes got rid of WL references (somewhere just removed, somewhere replaced by Routing)
- proxy404 REST endpoints URL change (was /wl now is /routing)

CI
https://bamboo.zion.sonatype.com/browse/NXO-OSSF44-1

Note: this pull once merged will need to be propagated to other codebases using proxy404...

### CODE REVIEW

**Comment by cstamas** [comment]
CI +1'd
https://bamboo.zion.sonatype.com/browse/NXO-OSSF44-1

**Comment by ifedorenko** [comment]
+1

I obviously did not check every single rename, but I trust whatever IDE you used to do the right thing (and javac to complain, if IDE did do something wrong)

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
CI +1'd
https://bamboo.zion.sonatype.com/browse/NXO-OSSF44-2

File: nexus-test/nexus-core-testsuite/src/test/java/core/routing/SmokeIT.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
  * Eclipse Foundation. All other trademarks are the property of their respective owners.
  */
-package core.whitelist;
+package core.routing;
```
Comment (jdillon) [inline nexus-test/nexus-core-testsuite/src/test/java/core/routing/SmokeIT.java:13]: different style of package name here still jarring (not expecting this to change in this pull btw, just while scanning quickly though I saw this again and stopped)

**Comment by jdillon** [comment]
más uno

File: nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/routing/internal/ManagerImpl.java [hunk 2]
```diff
@@ -158,7 +158,7 @@ public WLManagerImpl( final EventBus eventBus, final ApplicationStatusSource app
         this.remoteContentDiscoverer = checkNotNull( remoteContentDiscoverer );
         this.quickRemoteStrategy = checkNotNull( quickRemoteStrategy );
         this.executor =
-            new ScheduledThreadPoolExecutor( 5, new NexusThreadFactory( "wl", "WL-Updater" ),
+            new ScheduledThreadPoolExecutor( 5, new NexusThreadFactory( "ar", "AR-Updater" ),
```
Comment (ifedorenko) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/routing/internal/ManagerImpl.java:161]: should have been arrrr

**Pull request merged**